src/utils/fileReader/index.ts

```diff
@@ -34,22 +39,31 @@ const fileReaderFactory = (options: Partial<FileReaderOptions>) => {
           );
         }
 
-        const { name, type, size, lastModified } = file;
+        const { type, size } = file;
+        let name: File['name'] | undefined;
+        let lastModified: File['lastModified'] | undefined;
+
+        if (file instanceof File) {
+          name = file.name;
+          lastModified = file.lastModified;
+        }
+
         resolve({
           name,
           type,
           size,
           lastModified,
           uploadDate: Date.now(),
           data: evt.target.result,
+          file,
```

Comment:
Having access to the original input file is useful for thumbnail optimisation. (see `UserAvatarUploader: handleFileUpload`)